param/parameterized.py

```diff
@@ -1553,6 +1718,19 @@ def __init__(self,**params):
         self.param._setup_params(**params)
         object_count += 1
 
+        # TODO: should move to param namespace? (like _param_value
+        # etc)
+        self._param_subscribers = {}
+
+        # add watched dependencies
+        #
+        # TODO: This isn't great. Also note anything here will happen
+        # for every instantiation.
```

Comment:
This bit seems to be pretty expensive, it seems to take about 4x longer than before to instantiate a parameterized class.

Reply:
Did it get any better? I'd expect it to be a bit better than before (and minimal impact when there's no watching), but there are still improvements I plan to make (future PR).